src/libsystemd/sd-device/sd-device.c

```diff
@@ -1564,34 +1564,68 @@ _public_ const char *sd_device_get_property_next(sd_device *device, const char *
         return key;
 }
 
-static int device_sysattrs_read_all(sd_device *device) {
+static int device_sysattrs_read_all_internal(sd_device *device, const char *subdir) {
+        _cleanup_free_ char *path_dir = NULL;
         _cleanup_closedir_ DIR *dir = NULL;
-        const char *syspath;
         struct dirent *dent;
+        const char *syspath;
         int r;
 
-        assert(device);
-
-        if (device->sysattrs_read)
-                return 0;
-
         r = sd_device_get_syspath(device, &syspath);
         if (r < 0)
                 return r;
 
-        dir = opendir(syspath);
+        if (subdir) {
+                _cleanup_free_ char *p = NULL;
+                struct stat statbuf;
+
+                p = path_join(syspath, subdir, "uevent");
+                if (!p)
+                        return -ENOMEM;
+
+                if (lstat(p, &statbuf) >= 0)
+                        /* this is a child device, skipping */
+                        return 0;
```

Comment:
Nice!

Reply:
Hmm, actually, why stat this? Isn't `access(p, F_OK)` enough?

Reply:
@yuwata did you see this comment? I don't think we should do stat and throw away the result.

Reply:
Oh, sorry. I missed the comment. I will update so. Thank you for the reminder.